src/storage.rs

```diff
@@ -9,9 +9,14 @@ use crate::{
 
 use error_stack::ResultExt;
 
-use diesel_async::{pooled_connection::bb8::PooledConnection, AsyncPgConnection};
+#[cfg(feature = "postgres_ssl")]
+use diesel::{ConnectionError, ConnectionResult};
+
+#[cfg(feature = "postgres_ssl")]
+use futures_util::future::{BoxFuture, FutureExt};
```

Comment:
I don't think we need this dependency, BoxFuture is a type allias of probably can create something like this instead.

```rust
pub type BoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + Send + 'a>>;
```